clinica/iotools/bids_utils.py

```diff
@@ -485,25 +493,23 @@ def write_scans_tsv(bids_dir, bids_ids, scans_dict):
     import os
     from glob import glob
     from os import path
-
     import pandas as pd
 
     for bids_id in bids_ids:
-        # Create the file
         sessions_paths = glob(path.join(bids_dir, bids_id, "ses-*"))
         for session_path in sessions_paths:
-            scans_df = pd.DataFrame()
             session_name = session_path.split(os.sep)[-1]
-            tsv_name = bids_id + "_" + session_name + "_scans.tsv"
-
+            scans_df = pd.DataFrame()
+            # Create the file
+            tsv_name = "_".join([bids_id, session_name, "scans.tsv"])
             # If the file already exists, remove it
             if os.path.exists(path.join(bids_dir, bids_id, session_name, tsv_name)):
                 os.remove(path.join(bids_dir, bids_id, session_name, tsv_name))
 
             mod_available = glob(path.join(bids_dir, bids_id, session_name, "*"))
             for mod in mod_available:
                 mod_name = os.path.basename(mod)
-                files = glob(path.join(mod, "*"))
+                files = glob(path.join(mod, "*.nii.gz"))
```

Comment:
Why is this required?

Reply:
Unlike for example OASIS1 where the data extraction procedure results in folders and files with very standardized formats, the procedure under OASIS3 is not standardized at all with a messy tree folders. It was therefore necessary to perform a regex to retrieve only the nifti.

Reply:
I see, thanks for clarifying. Are we sure this change will not impact other converters which use this utiliy function?

Reply:
Normally, each directory should eventually contain a nifiti. I have done tests for OASIS1, I don't have the files (data-test) for the other converters. I thought the Jenkins automated tests checked that based for exemple on those in test/nonregression/test_run_iotools.py. If needed, we can ask @mdiazmel for the data-test files for other converters that I can run by hand.

Reply:
Agreed

Reply:
There are cases where we have got a sidecar JSON file (containing additional metadata as prescribed by the BIDS spec) alongside the Nifti that we would want to copy.

Reply:
Are there any files in OASIS-3 that we definitely don't want to copy?

Reply:
OK, are there JSON metadata files in other DBs? It seemed to me that OASIS1 was fairly standard and did not contain any, but maybe it was just the data-test I was provided. In my case, these json files are copied together with the nifti in converters/oasis3_to_bids (convert_images; line 138).

OASIS3 does contain a number of files that should not be copied (e.g., images of post-process analysis results, etc). To get an idea, a test dataset I used can be downloaded [here](https://owncloud.icm-institute.org/index.php/s/FvfVmXPwYJO0JNx). Some examples of input image files are in/unorganized (and reference outputs in ref/). (If needed, tests are done with the following command: pytest --verbose nonregression/test_run_iotools.py -k "test_run_Oasis3ToBids")

Reply:
I was about to ask whether you could share a link to some example data. I have requested access to OASIS-3 but it's still pending approval.

> OK, are there JSON metadata files as inputs in other DBs?

That may happen in newer datasets (the Harvard one comes to mind).

Another thing to consider is that other modalities may not produce nifti files, but some other native research format  (like bvec / bval files for DTI). Those would not be captured with your change.

I think what needs to be done is ensuring that you only copy the right files from OASIS-3 in a staging BIDS directory, and only then call this utility to write the corresponding scan files.

Reply:
Oops! my bad: I rerun the code line by line and at this step ... this bids procedure is called in the convert_clinical_data function of the converter ... which takes place after the convert_images step, the file ordering procedure!

So ... when one run write_scans_tsv from bids_util called by the converter ... the files are already ordered according to the bids standards. And there was indeed no need to parse the nifti paths.

That's why I removed my regex in the last commit. OFC, the OASIS1 and OASIS3 diff tests (expected vs observed) work.